src/main/java/rodain/inspection/SipContentDirectory.java

```diff
@@ -0,0 +1,145 @@
+package rodain.inspection;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+import javafx.event.EventHandler;
+import javafx.scene.control.TreeItem;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+import rodain.rules.TreeNode;
+import rodain.source.ui.items.SourceTreeItem;
+import rodain.source.ui.items.SourceTreeItemState;
+
+/**
+ * Created by adrapereira on 17-09-2015.
+ */
+public class SipContentDirectory extends TreeItem<Object> implements SourceTreeItem{
+    public static final Image folderCollapseImage = new Image(ClassLoader.getSystemResourceAsStream("icons/folder.png"));
+    public static final Image folderExpandImage = new Image(ClassLoader.getSystemResourceAsStream("icons/folder-open.png"));
+    private static final Comparator comparator = createComparator();
+    //this stores the full path to the file or directory
+    private String fullPath;
+    private TreeNode treeNode;
+
+    public SipContentDirectory(TreeNode treeNode) {
+        super(treeNode.getPath());
+        this.treeNode = treeNode;
+        this.fullPath = treeNode.getPath();
+        this.setGraphic(new ImageView(folderCollapseImage));
+
+        //set the value
+        if (!fullPath.endsWith(File.separator)) {
```

Comment:
Why not use Path in TreeNode, instead of saving the path as a String.
Also, don't use file separator:
> You might decide to use File.separator in place of a hard-coded separator character. However, this isn't a good idea either. NIO.2 supports multiple file systems, but File.separator is associated with the default file system.